newIDE/app/src/ProjectsStorage/ResourceFetcher/LocalResourceFetcher.spec.js

```diff
@@ -115,4 +118,57 @@ describe('LocalResourceFetcher', () => {
       { resourceName: 'MyResourceToDownload', error: expect.any(Error) },
     ]);
   });
+
+  it('automatically retries if a resource failed', async () => {
+    const project = makeTestProjectWithResourcesToDownload();
+
+    // Ensure just the files to be downloaded are listed
+    const resourceNames = LocalResourceFetcher.getResourcesToFetch(project);
+    expect(resourceNames).toEqual(['MyResourceToDownload']);
+
+    // Mock a failed download once, then successful
+    const onProgress = jest.fn();
+    mockFn(optionalRequire.mockFsExtra.ensureDir).mockImplementation(
+      async () => {}
+    );
+    mockFn(optionalRequire.mockFsExtra.existsSync).mockImplementation(
+      () => false
+    );
+    mockFn(
+      optionalRequire.mockElectron.ipcRenderer.invoke
+    ).mockImplementationOnce(() =>
+      Promise.reject(new Error('Fake download failure'))
+    );
+    mockFn(
+      optionalRequire.mockElectron.ipcRenderer.invoke
+    ).mockImplementationOnce(() => Promise.resolve());
```

Comment:
that's a super cool test :) 
from the doc, you can apparently chain `mockImplementationOnce()`
so:
```
mockFn(
      optionalRequire.mockElectron.ipcRenderer.invoke
    ).mockImplementationOnce(() =>
      Promise.reject(new Error('Fake download failure'))
    ).mockImplementationOnce(() => Promise.resolve());
```

Reply:
Done 👍